packages/buendia-networking/data/usr/share/buendia/config.d/10-networking

```diff
@@ -12,192 +12,223 @@
 
 set -e; . /usr/share/buendia/utils.sh
 
-# If we have a Wi-Fi interface, treat it as the "main" one. Otherwise rely on
-# the ethernet interface.
-if [ -z $NETWORKING_WIFI_INTERFACE ]; then
-  MAIN_INTERFACE=$NETWORKING_ETHERNET_INTERFACE
-else
-  MAIN_INTERFACE=$NETWORKING_WIFI_INTERFACE
-  USE_WIFI=1
-fi
+# Return true if a network interface can be found by name
+interface_exists () {
+    [ -n "$1" ] && ( ip a | grep -q " $1:" )
+}
+
+# Configure an interface in /etc/network/interfaces
+configure_interface () {
+    iface=$1
+    addr=$2
+
+    # Bring down the interface during reconfiguration, but don't fail if we can't.
+    ifdown $iface || true
```

Comment:
It's odd that this function brings the interface down but not up.  If it's the caller's responsibility to call `ifup` after, I think it should also be the caller's responsibility to call `ifdown` before.

Reply:
Agreed. I factored this out into a pair of separate functions, to be run before and after `/etc/network/interfaces` is modified.